keps/sig-api-machinery/0006-apply.md

```diff
@@ -194,9 +195,44 @@ atomic. That can be specified with `// +mapType=atomic` or `//
 +structType=atomic` respectively. They map to the same openapi extension:
 `"x-kubernetes-map-type": "atomic"`.
 
-#### Kubectl
+#### Clients
 
-##### Server-side Apply
+##### Apply Status Header
+
+Client side apply in kubectl determines if the applied object would result in any changes on the apiserver and provides the user with this information.
+
+To provide similar information with server-side apply, a dedicated header is set by the apiserver
+if a **successful request** caused changes to the existing resource.
+
+```txt
+X-Kubernetes-Operation-Success-Type: "changes-made" || "changes-unneeded"
```

Comment:
Since the below text aims at eventually making this more general, we need to ensure that this is either a) complete or b) future-proof (i.e., can add values without breaking clients).

I can think of ways of extending this, so it meets b), but I'm not sure it's obvious how that would work.

Can you state, in the text, how we'd e.g. go about adding a classification? Since we don't have many headers like this in our system, I don't think we have a common pattern. I guess the most obvious way is "leave this header untouched, add a second header that includes the possibility of the new classification." Another way is 

Alternatively, I can live with the current state if both @deads2k and @liggitt agree.

I'd consider, instead of `X-Kubernetes-Operation-Success-Type`, maybe `X-Kubernetes-Operation-Effect`?